runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TranslationUtils.java

```diff
@@ -346,13 +345,16 @@ public static void rejectStateAndTimers(DoFn<?, ?> doFn) {
    * @param coderMap - mapping between TupleTag and a coder
    * @return a pair function to convert value to bytes via coder
    */
-  public static PairFunction<Tuple2<TupleTag<?>, WindowedValue<?>>, TupleTag<?>, byte[]>
+  public static PairFunction<
+          Tuple2<TupleTag<?>, WindowedValue<?>>,
+          TupleTag<?>,
+          ValueAndCoderKryoSerializable<WindowedValue<?>>>
       getTupleTagEncodeFunction(final Map<TupleTag<?>, Coder<WindowedValue<?>>> coderMap) {
     return tuple2 -> {
       TupleTag<?> tupleTag = tuple2._1;
       WindowedValue<?> windowedValue = tuple2._2;
       return new Tuple2<>(
-          tupleTag, CoderHelpers.toByteArray(windowedValue, coderMap.get(tupleTag)));
+          tupleTag, new ValueAndCoderKryoSerializable<>(windowedValue, coderMap.get(tupleTag)));
```

Comment:
As stated previously we avoid to use Kryo serialization by default. See comment below about separating class functionality (lazily serializing) from Serialization.